workspaces/arborist/lib/query-selector-all.js

```diff
@@ -0,0 +1,523 @@
+'use strict'
+
+const { resolve } = require('path')
+const { parser, arrayDelimiter } = require('@npmcli/query')
+const localeCompare = require('@isaacs/string-locale-compare')('en')
+const npa = require('npm-package-arg')
+const minimatch = require('minimatch')
+const semver = require('semver')
+
+// handle results for parsed query asts, results are stored in a map that
+// has a key that points to each ast selector node and stores the resulting
+// array of arborist nodes as its value, that is essential to how we handle
+// multiple query selectors, e.g: `#a, #b, #c` <- 3 diff ast selector nodes
+class Results {
+  #results = null
+  #currentAstSelector = null
+
+  constructor (rootAstNode) {
+    this.#results = new Map()
+    this.#currentAstSelector = rootAstNode.nodes[0]
+  }
+
+  /* eslint-disable-next-line accessor-pairs */
+  set currentAstSelector (value) {
+    this.#currentAstSelector = value
+  }
+
+  get currentResult () {
+    return this.#results.get(this.#currentAstSelector)
+  }
+
+  set currentResult (value) {
+    this.#results.set(this.#currentAstSelector, value)
+  }
+
+  // when collecting results to a root astNode, we traverse the list of
+  // child selector nodes and collect all of their resulting arborist nodes
+  // into a single/flat Set of items, this ensures we also deduplicate items
+  collect (rootAstNode) {
+    const acc = new Set()
+    for (const n of rootAstNode.nodes) {
+      for (const node of this.#results.get(n)) {
+        acc.add(node)
+      }
+    }
+    return acc
```

Comment:
```suggestion
    return new Set(rootAstNode.nodes.flatMap(n => this.#results.get(n)))
```
altho flatMap might not be available in all supported nodes :-(

Reply:
It is https://node.green/#ES2019-features-Array-prototype--flat--flatMap--Array-prototype-flatMap